client/jetpack-cloud/sections/partner-portal/assign-license-step-progress/index.tsx

```diff
@@ -51,19 +49,25 @@ interface Props {
 	currentStep: StepKey;
 }
 
-export default function AssignLicenseStepProgress( { currentStep }: Props ): ReactElement {
+export default function AssignLicenseStepProgress( { currentStep }: Props ): ReactElement | null {
 	const translate = useTranslate();
-	const isAgency = useSelector( isAgencyUser );
-	const hasPaymentMethod = useSelector( hasValidPaymentMethod );
-	const paymentMethodRequired = isAgency && ! hasPaymentMethod;
+	const paymentMethodRequired = useSelector( isPaymentMethodRequired );
+	const sites = useSelector( getSites ).length;
 
-	const steps: Step[] = [
-		{ key: 'issueLicense', label: translate( 'Issue new license' ) },
-		...( paymentMethodRequired
-			? [ { key: 'addPaymentMethod', label: translate( 'Add Payment Method' ) } as Step ]
-			: [] ),
-		{ key: 'assignLicense', label: translate( 'Assign license' ) },
-	];
+	const steps: Step[] = [ { key: 'issueLicense', label: translate( 'Issue new license' ) } ];
+
+	if ( paymentMethodRequired ) {
+		steps.push( { key: 'addPaymentMethod', label: translate( 'Add Payment Method' ) } );
+	}
+
+	if ( sites > 0 ) {
+		steps.push( { key: 'assignLicense', label: translate( 'Assign license' ) } );
+	}
+
+	// Don't show the breadcrumbs if we have less than 2 as they are not very informative in this case.
+	if ( steps.length < 2 ) {
```

Comment:
This is 122638% subjective, but should we define this value as a const at the top of the function? The suggestion is meant to improve readability of the function when glancing through the code - but should just be skipped if there's any sort of counter argument.

Reply:
I'm not sure I understand which value you're referring to - the 2? This is about not showing steps if there's only one step.

Reply:
Yes, it's the `2` value 😄 
Some developers like having constants at the beginning of methods/functions to highlight later conditions (they believe it increases readability - but honestly... I lean more towards leaner functions instead 😅).

Let's forget it!